test/graph_compaction_tests.cc

```diff
@@ -368,15 +367,36 @@ namespace detail {
 		auto& ggen = ctx.get_graph_generator();
 		test_utils::mock_buffer_factory mbf(&tm, &ggen);
 		auto buf = mbf.create_buffer(range<1>(1));
-		for(int i = 0; i < 5; ++i) {
+		for(int i = 0; i < 4; ++i) {
 			test_utils::build_and_flush(
 			    ctx, test_utils::add_host_task(tm, on_master_node, [&](handler& cgh) { buf.get_access<access_mode::discard_write>(cgh, all{}); }));
 		}
 
 		// This must depend on the first horizon, not first_collective
-		const auto second_collective = test_utils::build_and_flush(ctx, test_utils::add_host_task(tm, experimental::collective, [&](handler& cgh) {}));
+		const auto second_collective = test_utils::build_and_flush(
+		    ctx, test_utils::add_host_task(tm, experimental::collective, [&](handler& cgh) { buf.get_access<access_mode::read>(cgh, all{}); }));
 
-		check_task_commands_depend_on_horizon_only(first_collective, second_collective, dependency_kind::ORDER_DEP, ctx);
+		const auto& inspector = ctx.get_inspector();
```

Comment:
What's up with this uglification change?

Reply:
I initially factored out this function because I assumed it could be used in another test, and it turned out that assumption was wrong. I don't have a preference to either version.